File: tabulate-core/src/test/java/io/github/voytech/tabulate/core/builder/fluent/support/OutputStreamTestOutputBinding.java
package io.github.voytech.tabulate.core.builder.fluent.support;

import io.github.voytech.tabulate.template.result.OutputBinding;
import org.jetbrains.annotations.NotNull;

import java.io.OutputStream;

public class OutputStreamTestOutputBinding implements OutputBinding<TestRenderingContext, OutputStream> {

    @NotNull
    @Override
    public Class<OutputStream> outputClass() {
        return OutputStream.class;
    }

    @Override
    public void setOutput(@NotNull TestRenderingContext renderingContext, OutputStream output) { }

    @Override
    public void flush() { }
}

File: tabulate-core/src/test/kotlin/io/github/voytech/tabulate/support/OutputStreamTestOutputBinding.kt
package io.github.voytech.tabulate.support

import io.github.voytech.tabulate.core.reify
import io.github.voytech.tabulate.core.result.OutputBinding
import java.io.OutputStream
import java.util.logging.Logger

class OutputStreamTestOutputBinding: OutputBinding<TestRenderingContext, OutputStream> {
    override fun outputClass(): Class<OutputStream> = reify()
    override fun setOutput(renderingContext: TestRenderingContext, output: OutputStream) {
        logger.info("This is fake implementation of ResultProvider flushing results into OutputStream")
    }

    override fun flush() {
        logger.info("This is fake implementation of ResultProvider flushing results into OutputStream")
    }

    companion object {
        val logger: Logger = Logger.getLogger(OutputStreamTestOutputBinding::class.java.name)
    }
}